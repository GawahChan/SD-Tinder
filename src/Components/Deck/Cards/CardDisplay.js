import React, { Component } from "react";
import PropTypes from "prop-types";
import {
  CardContents,
  CardOverlay,
  CardContent,
  CardButtonContainer,
  CardSubtitleContainer,
  Container
} from "./style";

import {
  CardInfoButton,
  CardLikedButton,
  CardDislikedButton,
  CardSuperlikedButton
} from "../../../Common/Styled/Icons.styles";
import { Header, Subtitle } from "../../../Common/Styled/Typography.styles";

class CardDisplay extends Component {
  constructor(props) {
    super(props);
    this.state = {
      actionType: "",
      displayBio: false
    };
  }

  receiveAction = (id, actionType) => {
    this.props.animateAction(id, actionType);
    this.setState({ actionType });
  };

  toggleBio = () => this.setState({ displayBio: !this.state.displayBio });

  render() {
    const overlay = this.state.actionType
      ? this.state.actionType.toUpperCase()
      : this.props.action.toUpperCase();

    return (
      <CardContents actionColour={overlay} img={this.props.stuntDouble.img}>
        <CardOverlay actionColour={overlay}>{overlay}</CardOverlay>
        <CardContent>
          <Container>
            <Header>
              {this.props.stuntDouble.name}, {this.props.stuntDouble.age}
            </Header>
            <CardSubtitleContainer>
              <Subtitle>{this.props.stuntDouble.location}</Subtitle>
              <CardInfoButton onTap={this.toggleBio} />
            </CardSubtitleContainer>
            {this.state.displayBio && (
              <Subtitle>{this.props.stuntDouble.bio}</Subtitle>
            )}
          </Container>
          <Container>
            <CardButtonContainer>
              <CardDislikedButton
                onClick={() =>
                  this.receiveAction(this.props.stuntDouble.id, "disliked")
                }
              />
              <CardSuperlikedButton
                onClick={() =>
                  this.receiveAction(this.props.stuntDouble.id, "superliked")
                }
              />
              <CardLikedButton
                onClick={() =>
                  this.receiveAction(this.props.stuntDouble.id, "liked")
                }
              />
            </CardButtonContainer>
          </Container>
        </CardContent>
      </CardContents>
    );
  }
}

CardDisplay.propTypes = {
  stuntDouble: PropTypes.object,
  animateAction: PropTypes.func,
  action: PropTypes.string
};
export default CardDisplay;
