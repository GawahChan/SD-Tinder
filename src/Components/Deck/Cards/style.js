import styled from 'styled-components';
import { motion } from 'framer-motion';

export const CardsContainer = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    min-height: 100vh;
`
// eslint-disable-next-line 
export const CardContainer = styled(motion.div).attrs(props => ({
    style: {
        boxShadow: props.actionColour === 'liked' ? '0px 0px 20px #15ff00' :
            props.actionColour === 'disliked' ? '0px 0px 20px #ff0000' :
                props.actionColour === 'superliked' ? '0 0 20px #02fdfd' : '0px 0px 10px #080808',

        border: props.actionColour === 'liked' ? '1px solid #15ff00' :
            props.actionColour === 'disliked' ? '1px solid #ff0000' :
                props.actionColour === 'superliked' ? '1px solid #02fdfd' : 'none'
    }
}))
    `
    position: absolute;
    border-radius: 10px;
    overflow: hidden;

    width: 320px;
    height: 500px;
        
    transition: box-shadow 0.5s, border 0.5s;
    will-change: transform;
`

export const CardContents = styled(motion.div)`
    width: 100%;
    height: 100%;

    display: flex;
    flex-direction: column;
    justify-content: flex-end; 

    background-image: ${props => {
        return (
            `linear-gradient(rgba(0,0,0,0), rgba(0,0,0,0), rgba(0,0,0,0.5), rgba(0,0,0,1)),
            url(${require(`../../../Common/Profile/${props.img}.jpg`)})`
        )

    }};
    background-position: top;
    background-size: cover;

    transition: box-shadow 0.5s;
    will-change: transform;
`

export const CardContent = styled.div`
    padding: 1rem;
    color: #DDDDDD;
`
// eslint-disable-next-line 
export const CardOverlay = styled.h1.attrs(props => ({
    style: {
        color: props.actionColour === 'LIKED' ? '#15ff00' :
            props.actionColour === 'DISLIKED' ? '#ff0000' :
                props.actionColour === 'SUPERLIKED' ? '#02fdfd' : 'black'
    }
}))
    `
    height: 30%;
    width: 100%;

    display: flex;
    justify-content: center;
    align-items: center;

    font-size: 3rem;
    font-weight: 500;

`
export const CardTitle = styled.h1`
    margin: 0;
    font-weight: 400;
    font-size: 2rem;
`

export const CardSubtitle = styled.div`
    display: flex;
    align-items: center;
    justify-content: space-between;
`

export const CardInfoButton = styled(motion.div)`
    background-image: url(${require(`../../../Common/Icons/info.svg`)});
    background-size: cover;

    height: 25px;
    width: 25px;
    cursor: pointer;
    outline: 0;
`

export const CardButtonContainer = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    margin: 5px; 
`

export const CardLikedButton = styled(motion.div)`
    background-image: url(${require(`../../../Common/Icons/liked.svg`)});
    background-size: cover;
    height: 50px;
    width: 50px;
    margin: 10px;
    cursor: pointer;
    outline: 0;
`

export const CardDislikedButton = styled(CardLikedButton)`
    background-image: url(${require(`../../../Common/Icons/disliked.svg`)});

`

export const CardSuperlikedButton = styled(CardLikedButton)`
    background-image: url(${require(`../../../Common/Icons/superliked.svg`)});

`