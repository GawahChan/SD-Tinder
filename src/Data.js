const Data = [
    {id: 0, name: 'Mila Kunis', age: '30', location: 'LA', bio: 'Actress' }
];

export default Data;