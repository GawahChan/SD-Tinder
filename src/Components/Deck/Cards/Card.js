import React, { Component } from 'react';
import { CardContainer } from './style';
import CardDisplay from './CardDisplay';

class Card extends Component {
    constructor(props) {
        super(props);
        this.state = {
            action: '',
            removeCard: false
        }
    }

    animateAction = (id, action) => {
        let handleCardAnimation =
            action === 'liked' ? { x: 400, opacity: 0 } :
                action === 'disliked' ? { x: -400, opacity: 0 } :
                    action === 'superliked' ? { y: -400, opacity: 0 } :
                        { x: 0, y: 0 }

        this.setState({ action, removeCard: handleCardAnimation })

        this.props.handleAction(id, action);
    }

    handleDrag = (info) => {
        let xPosition = info.point.x;
        let yPosition = info.point.y;

        let action =
            xPosition > 30 ? 'liked' :
                xPosition < -30 ? 'disliked' :
                    yPosition < -30 ? 'superliked' : '';

        this.setState({ action });
    }

    handleDragEnd = (id, info) => {
        let xPosition = info.point.x;
        let yPosition = info.point.y;

        return (
            xPosition > 80 ? this.animateAction(id, 'liked') :
                xPosition < -80 ? this.animateAction(id, 'disliked') :
                    yPosition < -80 ? this.animateAction(id, 'superliked') :
                        this.setState({ action: '' })
        );
    }
    render() {
        return (
            <CardContainer
                drag
                dragElastic={0.8}
                dragConstraints={{ top: 0, right: 0, bottom: 0, left: 0 }}
                onDrag={(e, info) => this.handleDrag(info)}
                onDragEnd={(e, info) => this.handleDragEnd(this.props.stuntDouble.id, info)}
                animate={this.state.removeCard}
                actionColour={this.state.action}
            >
                <CardDisplay
                    stuntDouble={this.props.stuntDouble}
                    animateAction={this.animateAction}
                    action={this.state.action}
                />
            </CardContainer >
        );
    }
}

export default Card;