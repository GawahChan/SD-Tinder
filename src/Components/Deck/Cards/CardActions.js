import React from 'react';

import {CardButtonContainer, CardLikedButton, CardDislikedButton, CardSuperlikedButton} from './style';

function CardActions({id, receiveAction}) {
    return (
        <CardButtonContainer>
            <CardDislikedButton onClick={() => receiveAction(id, 'disliked')}/>
            <CardSuperlikedButton onClick={() => receiveAction(id, 'superliked')} />
            <CardLikedButton onClick={() => receiveAction(id, 'liked')} />
        </CardButtonContainer>
    );
}

export default CardActions;