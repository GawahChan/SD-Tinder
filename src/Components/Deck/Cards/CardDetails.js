import React, { Component } from 'react';
import { CardTitle, CardSubtitle, CardInfoButton } from './style';

class CardDetails extends Component {
    constructor(props) {
        super(props);
        this.state = { displayBio: false }
    }

    toggleBio = () => this.setState({ displayBio: !this.state.displayBio })

    render() {
        const { stuntDouble } = this.props;
        return (
            <div>
                <CardTitle>
                    {stuntDouble.name}, {stuntDouble.age}
                </CardTitle>
                <CardSubtitle>
                    {stuntDouble.location}
                    <CardInfoButton onTap={this.toggleBio} />
                </CardSubtitle>
                {this.state.displayBio && <p>{stuntDouble.bio}</p>}
            </div >
        );
    }
}

export default CardDetails;