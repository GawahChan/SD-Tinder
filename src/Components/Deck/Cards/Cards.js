import React from 'react';
import Card from './Card';

import { CardsContainer } from './style';

function Cards(props) {
    return (
        <CardsContainer>
            {
                props.stuntDoubleList.map(stuntDouble => {
                    return (
                        <Card
                            key={stuntDouble.id}
                            stuntDouble={stuntDouble}
                            handleAction={props.handleAction}
                        />
                    );
                })
            }
        </CardsContainer>
    );
}

export default Cards;