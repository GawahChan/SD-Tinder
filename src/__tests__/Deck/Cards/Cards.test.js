import React from 'react';
import { shallow } from 'enzyme';
import Cards from '../../../Components/Deck/Cards/Cards';
import Data from '../../../Data';

describe('Cards Component', () => {
    let data = [Data[0]];

    let props = { stuntDoubleList: data }
    let cards = shallow(<Cards {...props} />)

    it('renders correctly', () => {
        expect(cards).toMatchSnapshot();
    });

    it('<Card /> Component is rendered when there is data in stuntDoubleList prop', () => {
        expect(cards.find('Card').exists()).toBe(true);
    });

    describe('when there is no data in stuntDoubleList prop', () => {
        let data = [];
        let props = { stuntDoubleList: data};
        let cards = shallow(<Cards {...props} />)

        it('Does NOT render <Card /> Component', () => {
            expect(cards.find('Card').exists()).toBe(false);
        });
    });

})
