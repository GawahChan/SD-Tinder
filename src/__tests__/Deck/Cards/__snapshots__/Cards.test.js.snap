// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards Component renders correctly 1`] = `
<styled.div>
  <Card
    key="0"
    stuntDouble={
      Object {
        "age": "28",
        "bio": "Recently finished stunt doubling for the movie 'Alita: Battle Angel' and am looking for a new role! Have stunt doubled for major films such as Wonder Woman (Gal Gadot), 'The Punisher' and 'Marvel's Agents of S.H.I.E.L.D'. If you're interested, swipe right and drop me a message!",
        "id": 0,
        "img": "Alicia-Vela-Bailey",
        "location": "LA, USA",
        "name": "Alicia Vela-Bailey",
      }
    }
  />
</styled.div>
`;
